### DISCUSSION TITLE
Feature additions

### DISCUSSION THREAD

**Discussion by rashil2000** [lptstr/winfetch#50] [Ideas]
Pasting the list from #29 here: 

- [x] Motherboard information
- [x] Multiple displays information
- [x] Multiple disks information
- [x] Process count
- [x] Processor load
- [x] Support for more Windows logos
- [x] App theme or system theme detection
- [x] Battery status
- [x] Locale (culture) information
- [x] Local IP, Public IP address
- [x] More colors in color bar
- [x] Graph-like metrics
- [x] Truncate output
- [ ] Songs, music player information

Graph-like implementation inspiration can be taken from [New-PercentageBar](https://github.com/rgel/PowerCLi/blob/master/Vi-Module/New-PercentageBar.ps1) :p

**Comment by kiedtl** [comment]
On 21-02-04 05:44 , Rashil Gandhi wrote:
> Graph-like implementation inspiration can be taken from [New-PercentageBar](https://github.com/rgel/PowerCLi/blob/master/Vi-Module/New-PercentageBar.ps1) :p

Ugh. Be careful when looking at implementations from other "competing" programs. License trouble is the last thing we need!

**Reply by rashil2000** [reply]
Hey @kiedtl ! 
Thanks for the heads up, I'll keep that in mind. 

How are you liking the changes so far?

**Reply by rashil2000** [reply]
> License trouble is the last thing we need!

BTW I just checked, New-PercentageBar is also under MIT license.

**Comment by kiedtl** [comment]
Going good!

**Comment by rashil2000** [comment]
@jcwillox regarding showing more colors in color bar, is there a way to print an extra line so that the rest 8 colors (the _bright_ variants) can be displayed?

Below is how neofetch shows it:
![image](https://user-images.githubusercontent.com/46838874/107139288-1a992680-6940-11eb-8d2c-419925b16dd0.png)

**Reply by jcwillox** [reply]
Yes we can, since PR #43, info functions can return an array of hashtables, one for each line it wants to display. Currently, the disk info function is the only one that uses that functionality.
```pwsh
function info_colorbar {
    return @(
        @{
            title   = ""
            content = ('{0}[0;40m{1}{0}[0;41m{1}{0}[0;42m{1}{0}[0;43m{1}' +
                '{0}[0;44m{1}{0}[0;45m{1}{0}[0;46m{1}{0}[0;47m{1}' +
                '{0}[0m') -f $e, '   '
        },
        @{
            title   = ""
            content = ('...') -f $e, '   '
        }
    )
}
```

**Reply by rashil2000** [reply]
Ah, thanks, I'll incorporate this into a commit to display all the colors.

**Comment by rashil2000** [comment]
@jcwillox I think tackling #13 and #38 before a new release is made would be good. Otherwise, we only have scoop as of now. I can maybe handle the PSGallery part, could you look into choco?

**Reply by rashil2000** [reply]
Okay, so here's the story so far...

We can no longer publish this script to PSGallery under the name posh-winfetch, since that name was already taken by the person who published it earlier (and has now deactivated his PSGalery account). The good part is, we don't have to - we can publish the script under any other name. winfetch was renamed posh-winfetch in this repository in this commit a49e691. I have reverted that change, since we can't keep renaming our script just to keep up with PSGallery.

We have to come up with a new name to publish it in PSGallery. I suggest lptstr-winfetch - it's simple and to-the-point.

**Reply by rashil2000** [reply]
I have temporarily published a test version - pwshfetch-test-1 - on PSGallery. You can test it out with `Install-Script pwshfetch-test-1`.

**Reply by jcwillox** [reply]
Well, you've just broken the scoop package... we should really avoid renaming core files as it can break several things which expect that file to be called `posh-winfetch.ps1`. I don't have any issue calling the file `winfetch.ps1` but we need to be careful about making these changes.

**Reply by rashil2000** [reply]
![image](https://user-images.githubusercontent.com/46838874/107848122-04470b00-6e17-11eb-8210-70cf13ccba06.png)

Could you please elaborate? The previous version installs fine here.

I was going to send a PR to scoop-extras anyways, to update the download URL, but I can't do that unless we make a release.

**Reply by jcwillox** [reply]
It would be good to add PowerShell and Chocolately packages but we want to do it right, as you've seen this has been attempted before and it ends up with the PowerShell package not getting updated or being removed. Ideally, you need to find a way that not only you can publish the package, and perhaps use Github actions to automatically publish to PSGallery when a new release is created.

I can maybe look into creating a Chocolatey package, but I have never used it, scoop is the only package manager for me (on windows at least). The choco package should also automatically update itself, without user intervention like scoop does.

**Reply by jcwillox** [reply]
Yes, what I meant is that the manifest will need to be updated, not a critical issue, but it could break anything that expects the file to be called `posh-winfetch.ps1`, so I'll need to document this in the release notes.

Actually, you shouldn't need us to create a new release just change the `autoupdate` URL, when scoop detects a new version it will use the new URL

**Reply by rashil2000** [reply]
>perhaps use Github actions to automatically publish to PSGallery when a new release is created

Unfortunately, this'll be a slight hassle. @kiedtl will have to create a Microsoft account here, get the API Key, add the key to his GitHub secrets, and then write a custom Actions workflow to publish (since the name of the script is not the same as the one in the repo).

For now, manually doing it is super easy. Every time a release is made, I'd just need to do 
```powershell
Copy-Item winfetch.ps1 lptstr-winfetch.ps1
Publish-Script -Path lptstr-winfetch.ps1 -NuGetApiKey $MyKey
```

**Reply by rashil2000** [reply]
lukesampson/scoop-extras#5678

**Reply by jcwillox** [reply]
Sure that's a good enough option, but what if we want to switch to Github actions in the future will we have to rename the PowerShell package again? And what if you decided to stop maintaining winfetch? I'm not saying we can't do it that way, I just want to make sure we've thought about the potential issues.

**Reply by rashil2000** [reply]
> And what if you decided to stop maintaining winfetch?

To be honest, I'd want @kiedtl to run the aforementioned two commands himself, since those don't require you to be on Windows to do (and if it's not much trouble :))

I haven't gone ahead with publishing the actual script just because of this same reason.

**Reply by kiedtl** [reply]
> To be honest, I'd want @kiedtl to run the aforementioned two commands himself, since those don't require you to be on Windows to do (and if it's not much trouble :))

I mean, I can go ahead and create an MSFT account and all... I just can't give any guarantees as to when I'll end up doing it. I might end up doing it tomorrow, or next year.

**Reply by rashil2000** [reply]
The problem with setting up GH actions is also the same, if I do it, it'll use my GitHub actions, secrets, MSFT account and all, and if we change maintainers we'll have to start over again.

@kiedtl if and when you decide to do it (PSGallery can certainly wait, because we have Scoop (maybe choco too) already), it should be ideally be a one-time task of setting up the auto-publishing workflow.

**Reply by jcwillox** [reply]
Do you know if it's possible to publish a script to PSGallery under a different name to the actual script file, it would be better if users could type `winfetch` instead of `lptstr-winfetch`? It doesn't seem like they can.

**Reply by rashil2000** [reply]
No, both names should be the same. Although you can suggest the users (in the wiki) to setup an alias.

**Reply by jcwillox** [reply]
That's all good

@kiedtl I'm happy to create the workflow for publishing to PSGallery, but would you be able to obtain and add the `NuGetApiKey` to the Github secrets. Alternatively, I can create an account for winfetch and send you the details.

**Reply by rashil2000** [reply]
@jcwillox Should I add [this](https://github.com/lptstr/winfetch/issues/13#issuecomment-788792589) to the wiki, for the time being?

**Reply by jcwillox** [reply]
I wouldn't add it as an actual install method, maybe something more like, "there is a temporary PSGallery package available...". Scoop is still the only "officially" supported way to install winfetch, aside from just downloading it. If someone just wants a way to test winfetch then the direct download is still a good option.

Does PSGallery provide organisational accounts or do you just have to create a normal Microsoft account?

**Reply by rashil2000** [reply]
>maybe something more like, "there is a temporary PSGallery package available..."

Yes, that's what I meant.

>Does PSGallery provide organisational accounts or do you just have to create a normal Microsoft account?

If there's a way to create one, I'm not aware of it. From what I have seen, it just gives you a normal Microsoft account Sign In option. Now, that account might be a personal account, or an org account.

**Comment by rashil2000** [comment]
I just made a Reddit [post](https://www.reddit.com/r/PowerShell/comments/mav1qb/winfetch_200_released_new_segments_better_speed/) on r/PowerShell celebrating the new release!
